sources/settings-extensions/ntp/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+/// The ntp settings can be used to specify time servers with which to synchronize the instance's
+/// clock.
+use bottlerocket_settings_sdk::{GenerateResult, LinearlyMigrateable, NoMigration, SettingsModel};
+use model_derive::model;
+use modeled_types::Url;
+use std::convert::Infallible;
+
+#[model(impl_default = true)]
+pub struct NtpSettingsV1 {
+    time_servers: Vec<Url>,
+}
+
+type Result<T> = std::result::Result<T, Infallible>;
+
+impl SettingsModel for NtpSettingsV1 {
+    /// the `model` macro makes every field of the `NtpSettingsV1` struct an `Option`, so we can use
+    /// the type as its own `PartialKind`.
+    type PartialKind = Self;
+    type ErrorKind = Infallible;
+
+    fn get_version() -> &'static str {
+        "v1"
+    }
+
+    fn set(_current_value: Option<Self>, _target: Self) -> Result<()> {
+        // TODO: add additional validation -- legal for this to be set to any URL or do we need to
+        // check that it is a valid time server?
+        Ok(())
+    }
+
+    fn generate(
+        existing_partial: Option<Self::PartialKind>,
+        _dependent_settings: Option<serde_json::Value>,
+    ) -> Result<GenerateResult<Self::PartialKind, Self>> {
+        Ok(GenerateResult::Complete(
+            existing_partial.unwrap_or_default(),
+        ))
+    }
+
+    fn validate(_value: Self, _validated_settings: Option<serde_json::Value>) -> Result<()> {
+        // TODO: add additional validation -- legal for this to be set to any URL or do we need to
+        // check that it is a valid time server?
```

Comment:
We tend not to rely on external services when validating settings as long as they are the same shape, and I think that's the right move. It would be too easy for a transient error to cause unnecessary failure at scale.